// @sanity/desk-tool/lib/pane/Editor/Validation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Validation = Validation;

var _react = _interopRequireDefault(require("react"));

var _reactTippy = require("react-tippy");

var _schema = _interopRequireDefault(require("part:@sanity/base/schema"));

var _default = _interopRequireDefault(require("part:@sanity/components/buttons/default"));

var _list = _interopRequireDefault(require("part:@sanity/components/validation/list"));

var _chevronDownIcon = _interopRequireDefault(require("part:@sanity/base/chevron-down-icon"));

var _warningIcon = _interopRequireDefault(require("part:@sanity/base/warning-icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Validation(props) {
  var onCloseValidationResults = props.onCloseValidationResults,
      onFocus = props.onFocus,
      onToggleValidationResults = props.onToggleValidationResults,
      showValidationTooltip = props.showValidationTooltip,
      type = props.type,
      markers = props.markers;
  var validation = markers.filter(marker => marker.type === 'validation');
  var errors = validation.filter(marker => marker.level === 'error');
  var warnings = validation.filter(marker => marker.level === 'warning');

  if (errors.length === 0 && warnings.length === 0) {
    return null;
  }

  return /*#__PURE__*/_react.default.createElement(_reactTippy.Tooltip, {
    arrow: true,
    theme: "light",
    trigger: "click",
    position: "bottom",
    interactive: true,
    duration: 100,
    open: showValidationTooltip,
    onRequestClose: onCloseValidationResults,
    html: /*#__PURE__*/_react.default.createElement(_list.default, {
      truncate: true,
      markers: validation,
      showLink: true,
      isOpen: showValidationTooltip,
      documentType: _schema.default.get(type),
      onClose: onCloseValidationResults,
      onFocus: onFocus
    })
  }, /*#__PURE__*/_react.default.createElement(_default.default, {
    color: errors.length > 0 ? 'danger' : 'warning',
    bleed: true,
    icon: _warningIcon.default,
    padding: "small",
    onClick: onToggleValidationResults
  }, errors.length + warnings.length, /*#__PURE__*/_react.default.createElement("span", {
    style: {
      paddingLeft: '0.5em',
      display: 'flex'
    }
  }, /*#__PURE__*/_react.default.createElement(_chevronDownIcon.default, null))));
}
