// @sanity/default-layout/lib/components/CurrentVersionsDialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _default2 = _interopRequireDefault(require("part:@sanity/components/dialogs/default"));

var _content = _interopRequireDefault(require("part:@sanity/components/dialogs/content"));

var _UpdateNotifierDialog = _interopRequireDefault(require("./styles/UpdateNotifierDialog.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class CurrentVersionsDialog extends _react.Component {
  renderTable() {
    var versions = this.props.versions;
    return /*#__PURE__*/_react.default.createElement("table", {
      className: _UpdateNotifierDialog.default.versionsTable
    }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("th", null, "Module"), /*#__PURE__*/_react.default.createElement("th", null, "Installed"), /*#__PURE__*/_react.default.createElement("th", null, "Latest"))), /*#__PURE__*/_react.default.createElement("tbody", null, Object.keys(versions).map(pkgName => /*#__PURE__*/_react.default.createElement("tr", {
      key: pkgName
    }, /*#__PURE__*/_react.default.createElement("td", null, pkgName), /*#__PURE__*/_react.default.createElement("td", null, versions[pkgName]), /*#__PURE__*/_react.default.createElement("td", null, versions[pkgName])))));
  }

  render() {
    var onClose = this.props.onClose;
    return /*#__PURE__*/_react.default.createElement(_default2.default, {
      isOpen: true,
      onClose: onClose,
      onClickOutside: onClose
    }, /*#__PURE__*/_react.default.createElement(_content.default, {
      size: "medium",
      padding: "large"
    }, /*#__PURE__*/_react.default.createElement("div", {
      className: _UpdateNotifierDialog.default.content
    }, /*#__PURE__*/_react.default.createElement("h2", {
      className: _UpdateNotifierDialog.default.dialogHeading
    }, "Studio is up to date"), /*#__PURE__*/_react.default.createElement("p", null, "Your Studio is running on the latest updates."), /*#__PURE__*/_react.default.createElement("details", {
      className: _UpdateNotifierDialog.default.details
    }, /*#__PURE__*/_react.default.createElement("summary", {
      className: _UpdateNotifierDialog.default.summary
    }, "Show all installed modules"), this.renderTable()))));
  }

}

_defineProperty(CurrentVersionsDialog, "propTypes", {
  onClose: _propTypes.default.func.isRequired,
  versions: _propTypes.default.objectOf(_propTypes.default.string)
});

_defineProperty(CurrentVersionsDialog, "defaultProps", {
  versions: []
});

var _default = CurrentVersionsDialog;
exports.default = _default;
