// @sanity/default-layout/lib/components/UpdateNotifierDialog.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _default2 = _interopRequireDefault(require("part:@sanity/components/dialogs/default"));

var _content = _interopRequireDefault(require("part:@sanity/components/dialogs/content"));

var _UpdateNotifierDialog = _interopRequireDefault(require("./styles/UpdateNotifierDialog.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var upperFirst = str => "".concat(str.slice(0, 1).toUpperCase()).concat(str.slice(1));

class UpdateNotifierDialog extends _react.Component {
  renderTable() {
    var outdated = this.props.outdated;
    return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("table", {
      className: _UpdateNotifierDialog.default.versionsTable
    }, /*#__PURE__*/_react.default.createElement("thead", null, /*#__PURE__*/_react.default.createElement("tr", null, /*#__PURE__*/_react.default.createElement("th", null, "Module"), /*#__PURE__*/_react.default.createElement("th", null, "Installed"), /*#__PURE__*/_react.default.createElement("th", null, "Latest"), /*#__PURE__*/_react.default.createElement("th", null, "Importance"))), /*#__PURE__*/_react.default.createElement("tbody", null, outdated.map(pkg => /*#__PURE__*/_react.default.createElement("tr", {
      key: pkg.name
    }, /*#__PURE__*/_react.default.createElement("td", null, pkg.name), /*#__PURE__*/_react.default.createElement("td", null, pkg.version), /*#__PURE__*/_react.default.createElement("td", null, pkg.latest), /*#__PURE__*/_react.default.createElement("td", null, upperFirst(pkg.severity || 'low')))))), /*#__PURE__*/_react.default.createElement("div", {
      className: _UpdateNotifierDialog.default.upgradeText
    }, /*#__PURE__*/_react.default.createElement("p", null, "To upgrade, run the ", /*#__PURE__*/_react.default.createElement("a", {
      href: "https://www.sanity.io/docs/reference/cli"
    }, "Sanity CLI"), ' ', "upgrade command in your project folder from a terminal."), /*#__PURE__*/_react.default.createElement("code", {
      className: _UpdateNotifierDialog.default.code
    }, "sanity upgrade")));
  }

  renderContactDeveloper() {
    var severity = this.props.severity;
    return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement("p", null, "This Studio is now outdated.", ' ', severity === 'high' ? 'Please get in touch with your developers and ask them to upgrade it for you.' : 'Consider getting in touch with your developers and ask them to upgrade it for you.'), /*#__PURE__*/_react.default.createElement("details", {
      className: _UpdateNotifierDialog.default.details
    }, /*#__PURE__*/_react.default.createElement("summary", {
      className: _UpdateNotifierDialog.default.summary
    }, "Developer info"), this.renderTable()));
  }

  render() {
    var _this$props = this.props,
        severity = _this$props.severity,
        onClose = _this$props.onClose;
    return /*#__PURE__*/_react.default.createElement(_default2.default, {
      isOpen: true,
      onClose: onClose,
      onClickOutside: onClose
    }, /*#__PURE__*/_react.default.createElement(_content.default, {
      size: "medium",
      padding: "large"
    }, /*#__PURE__*/_react.default.createElement("h2", {
      className: _UpdateNotifierDialog.default.dialogHeading
    }, severity === 'low' ? 'Upgrades available' : 'Studio is outdated'), __DEV__ && /*#__PURE__*/_react.default.createElement("p", null, "This Studio is no longer up to date", ' ', severity === 'high' ? 'and should be upgraded.' : 'and can be upgraded.'), __DEV__ ? this.renderTable() : this.renderContactDeveloper()));
  }

}

_defineProperty(UpdateNotifierDialog, "propTypes", {
  onClose: _propTypes.default.func.isRequired,
  severity: _propTypes.default.string.isRequired,
  outdated: _propTypes.default.arrayOf(_propTypes.default.shape({
    name: _propTypes.default.string,
    latest: _propTypes.default.string,
    severity: _propTypes.default.string
  }))
});

_defineProperty(UpdateNotifierDialog, "defaultProps", {
  outdated: []
});

var _default = UpdateNotifierDialog;
exports.default = _default;
